aws/adapter.go

```diff
@@ -188,9 +200,58 @@ func (a *Adapter) InstanceID() string {
 	return a.manifest.instance.id
 }
 
-// AutoScalingGroupName returns the name of the Auto Scaling Group the current node belongs to
-func (a *Adapter) AutoScalingGroupName() string {
-	return a.manifest.autoScalingGroup.name
+// AutoScalingGroupNames returns names of the Auto Scaling Groups that
+// kubernetes nodes belong to.
+func (a *Adapter) AutoScalingGroupNames() []string {
+	result := make([]string, len(a.autoScalingGroups))
+	i := 0
+	for name := range a.autoScalingGroups {
+		result[i] = name
+		i++
+	}
+	return result
+}
+
+// SingleInstances returns list of IDs of instances that do not belong to any
+// Auto Scaling Group and should be managed manually.
+func (a *Adapter) SingleInstances() []string {
+	instances := make([]string, 0, len(a.singleInstances))
+	for id := range a.singleInstances {
+		instances = append(instances, id)
+	}
+	return instances
+}
+
+// RunningSingleInstances returns list of IDs of running instances that do
+// not belong to any Auto Scaling Group and should be managed manually.
+func (a *Adapter) RunningSingleInstances() []string {
+	instances := make([]string, 0, len(a.singleInstances))
+	for id, details := range a.singleInstances {
+		if details.running {
+			instances = append(instances, id)
+		}
+	}
+	return instances
+}
+
+// ObsoleteSingleInstances returns list of IDs of instances that should be deregistered
+// from all Target Groups.
+func (a *Adapter) ObsoleteSingleInstances() []string {
+	return a.obsoleteInstances
+}
+
+// Get number of instances in cache.
+func (a *Adapter) CachedInstances() int {
+	return len(a.instancesDetails)
+}
+
+// Get EC2 filters that are used to filter instances that are loaded using DescribeInstances.
+func (a *Adapter) FiltersString() string {
```

Comment:
we could use the value receiver here.

Reply:
Would not it have impact on performance? If we use value receiver here it will mean that each call to this method will make copy of Adapter struct, no?